projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -1552,61 +1553,77 @@ private function get_outline_styles( $form_style = 'outlined' ) {
-			$legacy_border_radius = is_numeric( $legacy_border_radius ) ? $legacy_border_radius . 'px' : $legacy_border_radius;
-
-			$border_size = $legacy_border_size ??
-				$outline_styles['border']['top']['width'] ??
-				$global_styles['width'] ??
-				$global_styles['top']['width'];
-
-			$border_radius = $legacy_border_radius ??
-				$global_styles['radius'];
-
-			$css_vars = $border_size ? '--jetpack--contact-form--border-size: ' . $border_size . ';' : '';
-			// Check if border radius is split or a single value.
-			if ( is_array( $border_radius ) ) {
-				// If corner radii are set on the top-left or bottom-left of the block, take the maximum of the two.
-				// We check the left side due to writing direction—this variable is used to offset text.
-				// TODO: this should factor in RTL languages.
-				$css_vars .= $border_radius ? '--jetpack--contact-form--border-radius: max(' . $border_radius['topLeft'] . ',' . $border_radius['bottomLeft'] . ');' : '';
-			} elseif ( isset( $border_radius ) ) {
-				$css_vars .= $border_radius ? '--jetpack--contact-form--border-radius: ' . $border_radius . ';' : '';
-			}
+		$outline_styles = ! empty( $outline_styles ) ? json_decode( html_entity_decode( $outline_styles, ENT_COMPAT ), true ) : array();
+		$input_styles   = $this->get_attribute( 'inputstyles' );
 
-			if ( 'outlined' === $form_style ) {
-				$css_vars .= '--jetpack--contact-form--notch-width: max(var(--jetpack--contact-form--input-padding-left, 16px), var(--jetpack--contact-form--border-radius));';
-			} elseif ( 'animated' === $form_style ) {
-				$legacy_border_left_size = $outline_styles['border']['width'] ?? null;
-				$legacy_border_left_size = is_numeric( $legacy_border_left_size ) ? $legacy_border_left_size . 'px' : $legacy_border_left_size;
+		if ( ! empty( $input_styles ) ) {
+			$style_attrs = $input_styles;
+		}
 
-				$border_left_size = $legacy_border_left_size ??
-				$outline_styles['border']['left']['width'] ??
-				$global_styles['width'] ??
-				$global_styles['left']['width'];
+		$global_styles = wp_get_global_styles(
+			array( 'border' ),
+			array(
+				'block_name' => $block_name,
+				'transforms' => array( 'resolve-variables' ),
+			)
+		);
 
-				$css_vars .= "--jetpack--contact-form--left-offset: calc(var(--jetpack--contact-form--input-padding-left, 16px) + {$border_left_size});";
-				$css_vars .= '--jetpack--contact-form--label-left: max(var(--jetpack--contact-form--left-offset), var(--jetpack--contact-form--border-radius));';
-				$css_vars .= "--jetpack--contact-form--field-padding: calc(var(--jetpack--contact-form--label-left) - {$border_left_size});";
-			}
+		// The `borderwidth` attribute contains the border value that forms used before the migration to global styles.
+		// Any old forms saved in a post will still use this attribute, so it needs to be factored into the css vars for border
+		// to properly support backwards compatibility. Use `isset` to check instead of `empty` as `'0'` is a valid width value
```

Comment:
I'm seeing px units with no values :thinking_face:
<img width="1249" alt="Screenshot 2025-05-20 at 12 50 58 pm" src="https://github.com/user-attachments/assets/579abc31-d139-450b-a621-e43b3874fa6c" />

I think this is because `""` will return true for `isset`

```php
$i_am_empty = "";
var_dump(isset($i_am_empty));

// bool(true)

```

Should we just check for empty and zero? E.g., 

`! empty( $border_width_attribute ) || $border_width_attribute === '0'`

Reply:
Done in [f1be8bb](https://github.com/Automattic/jetpack/pull/43469/commits/f1be8bb93cfb8adee1c1f378982f4d2212f1f218)